zip-0155.rst

```diff
@@ -0,0 +1,246 @@
+::
+
+  ZIP: 155
+  Title: addrv2 message
+  Owners: Daira Hopwood <daira@electriccoin.co>
+  Credits: Wladimir J. van der Laan
+           Zancas Wilcox
+  Status: Proposed
+  Category: Network
+  Created: 2021-08-13
+  License: BSD-2-Clause
+  Discussions-To: <https://github.com/zcash/zips/issues/542>
+  Pull-Request: <https://github.com/zcash/zips/pull/543>
+
+
+Terminology
+===========
+
+The key words "MUST", "MUST NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED" and "MAY"
+in this document are to be interpreted as described in RFC 2119. [#RFC2119]_
+
+The term "network upgrade" in this document is to be interpreted as described in
+ZIP 200. [#zip-0200]_
+
+The terms "Testnet" and "Mainnet" are to be interpreted as described in
+section 3.11 of the Zcash Protocol Specification [#protocol-networks]_.
+
+"P2P network" means the Zcash peer-to-peer network.
+
+
+Abstract
+========
+
+This document proposes a new P2P message to gossip longer node addresses over the
+P2P network. This is required to support new-generation Onion addresses, I2P, and
+potentially other networks that have longer endpoint addresses than fit in the 128
+bits of the current ``addr`` message.
+
+
+Motivation
+==========
+
+Tor v3 hidden services are part of the stable release of Tor since version 0.3.2.9.
+They have various advantages compared to the old hidden services, among which better
```

Comment:
```suggestion
Tor v3 onion services are part of the stable release of Tor since version 0.3.2.9.
They have various advantages compared to the old v2 onion services, among which better
```